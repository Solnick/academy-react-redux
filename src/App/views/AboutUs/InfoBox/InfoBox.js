import React from 'react';
import PropTypes from 'prop-types';
import Tile from '../../../components/Tile/Tile';
import styles from './infoBox.module.scss';

const InfoBox = ({ text, title, img }) => (
  <Tile>
    <div className={styles.infoBoxWrapper}>
      <img src={img} alt={title} />
      <div className={styles.textWrapper}>
        <h2>{title}</h2>
        <p>{text}</p>
      </div>
    </div>
  </Tile>
);

InfoBox.propTypes = {
  text: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
};

export default InfoBox;
