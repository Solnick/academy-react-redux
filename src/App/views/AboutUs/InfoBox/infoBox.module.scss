.infoBoxWrapper {
  width: min-content;
  img {
    width: inherit;
  }
  .textWrapper {
    line-height: 20px;
    margin: 20px;
    h2 {
      font-size: 1.75rem;
      font-weight: 400;
    }
    p {
      margin-top: 15px
    }
  }
}