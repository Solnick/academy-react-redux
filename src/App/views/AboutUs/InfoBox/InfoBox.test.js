import React from 'react';
import { shallow } from 'enzyme';
import InfoBox from './InfoBox';
import testImage from '../../../assets/images/infoBox1.png';

describe('InfoBox component', () => {
  describe('when provided with props', () => {
    const props = {
      img: testImage,
      text: 'text',
      title: 'title',
    };
    it('contains h2 element with title', () => {
      const infoBoxInstance = shallow(<InfoBox
        {...props}
      />);
      expect(infoBoxInstance.containsMatchingElement(<h2>{props.title}</h2>));
    });
    it('contains paragraph element with title', () => {
      const infoBoxInstance = shallow(<InfoBox
        {...props}
      />);
      expect(infoBoxInstance.containsMatchingElement(<p>{props.text}</p>));
    });
    it('contains paragraph element with title', () => {
      const infoBoxInstance = shallow(<InfoBox
        {...props}
      />);
      expect(infoBoxInstance.containsMatchingElement(<img src={props.img} alt={props.img} />));
    });
    it('should render correctly', () => {
      const infoBoxInstance = shallow(<InfoBox
        {...props}
      />);
      expect(infoBoxInstance).toMatchSnapshot();
    });
  });
});
