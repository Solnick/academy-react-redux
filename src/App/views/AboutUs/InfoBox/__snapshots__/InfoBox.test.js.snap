// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBox component when provided with props should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InfoBox
    img="infoBox1.png"
    text="text"
    title="title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "center": false,
      "children": <div
        className="infoBoxWrapper"
      >
        <img
          alt="title"
          src="infoBox1.png"
        />
        <div
          className="textWrapper"
        >
          <h2>
            title
          </h2>
          <p>
            text
          </p>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="title"
            src="infoBox1.png"
          />,
          <div
            className="textWrapper"
          >
            <h2>
              title
            </h2>
            <p>
              text
            </p>
          </div>,
        ],
        "className": "infoBoxWrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "title",
            "src": "infoBox1.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                title
              </h2>,
              <p>
                text
              </p>,
            ],
            "className": "textWrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "title",
              },
              "ref": null,
              "rendered": "title",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "text",
              },
              "ref": null,
              "rendered": "text",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "center": false,
        "children": <div
          className="infoBoxWrapper"
        >
          <img
            alt="title"
            src="infoBox1.png"
          />
          <div
            className="textWrapper"
          >
            <h2>
              title
            </h2>
            <p>
              text
            </p>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="title"
              src="infoBox1.png"
            />,
            <div
              className="textWrapper"
            >
              <h2>
                title
              </h2>
              <p>
                text
              </p>
            </div>,
          ],
          "className": "infoBoxWrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "title",
              "src": "infoBox1.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  title
                </h2>,
                <p>
                  text
                </p>,
              ],
              "className": "textWrapper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "title",
                },
                "ref": null,
                "rendered": "title",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "text",
                },
                "ref": null,
                "rendered": "text",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
